src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskCancelByIdControllerTest.java

```diff
@@ -0,0 +1,115 @@
+            .body("status", equalTo(HttpStatus.NOT_FOUND.value()))
+            .body("message", equalTo(String.format(
+                LOG_MSG_THERE_WAS_A_PROBLEM_FETCHING_THE_VARIABLES_FOR_TASK,
+                nonExistentTaskId
+            )));
+    }
+
+    @Test
+    public void should_return_a_401_when_the_user_did_not_have_any_roles() {
+
+        Map<String, String> task = common.setupTaskAndRetrieveIds(Common.TRIBUNAL_CASEWORKER_PERMISSIONS);
+        var taskId = task.get("taskId");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            authorizationHeadersProvider.getLawFirmBAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.UNAUTHORIZED.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.UNAUTHORIZED.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.UNAUTHORIZED.value()))
+            .body("message", equalTo("User did not have sufficient permissions to perform this action"));
+
+        common.cleanUpTask(taskId, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_403_when_the_user_did_not_have_sufficient_jurisdiction_did_not_match() {
+        Map<String, String> task = common.setupTaskAndRetrieveIdsWithCustomVariable(JURISDICTION, "SSCS");
+        var taskId = task.get("taskId");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.FORBIDDEN.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("timestamp", lessThanOrEqualTo(LocalDateTime.now()
+                                                     .format(DateTimeFormatter.ofPattern(DATE_TIME_FORMAT))))
+            .body("error", equalTo(HttpStatus.FORBIDDEN.getReasonPhrase()))
+            .body("status", equalTo(HttpStatus.FORBIDDEN.value()))
+            .body("message", equalTo(
+                format("User did not have sufficient permissions to cancel task with id: %s", taskId)
+            ));
+
+        common.cleanUpTask(taskId, REASON_COMPLETED);
+    }
+
+    @Test
+    public void should_return_a_204_when_cancelling_a_task_by_id() {
+        Map<String, String> task = common.setupTaskAndRetrieveIds(Common.TRIBUNAL_CASEWORKER_PERMISSIONS);
+        var taskId = task.get("taskId");
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            authorizationHeadersProvider.getTribunalCaseworkerAAuthorization()
+        );
+
```

Comment:
AC2 will also need to be covered. It's similar to other endpoints. I.e. testing CASE ROLE/CASE ID after we set up/mock full ccd integration. (We can discuss- it will be a separate ticket).

AC5 -> Then the endpoint will return a 5XX range HTTP error (role assignment service error); This can be in an integration test. We can also discuss.

Reply:
AC2 - yes it can be part of the ticket 225.
AC5 - I think we got a story for writing integration tests to handle 500, so this can be part of that ticket.

Reply:
AC5- new ticket created. 
https://tools.hmcts.net/jira/browse/RWA-281

Reply:
As a general principle, we should cover all the ACs as part of the feature ticket as this should be incorporated into the points. Otherwise, if tests are written after the feature is merged, there is an opportunity for a bug to be merged.  

In this case we can do the work for 225 separately since the Jira is already there.  281 should have been done here so maybe merge this and work on that next.

Reply:
sure.